File: SatelliteAndroid/satellite/src/test/java/yuni/library/satellite/LocationServiceMock.java
package yuni.library.satellite;

public class LocationServiceMock implements LocationService {

    private boolean mStarted;
    private Satellite.Listener mListener;
    private Satellite.Options mOptions;

    private FakeClock mClock;
    private Location mLastLocation;

    public LocationServiceMock(FakeClock clock) {
        mClock = clock;
    }

    @Override
    public Location getLastLocation() {
        return mLastLocation;
    }

    @Override
    public boolean isStarted() {
        return mStarted;
    }

    @Override
    public void start(Satellite.Options options) {
        mStarted = true;
    }

    @Override
    public void stop() {
        mStarted = false;
    }

    @Override
    public void setListener(Satellite.Listener listener) {
        mListener = listener;
    }

    @Override
    public void updateOptions(Satellite.Options options) {
        mOptions = options;
    }

    @Override
    public void getOnce(Satellite.Options options, Satellite.Listener listener) {
        Location location = new Location();
        location.setTime(mClock.now);
        listener.onLocationChanged(location);
    }

    public void fireLocaiton() {
        if (!isStarted()) {
            return;
        }
        Location location = new Location();
        location.setTime(mClock.now);
        mLastLocation = location;
        mListener.onLocationChanged(location);
    }

}

File: SatelliteIOS/Example/Tests/LocationServiceMock.swift
//
//  LocationServiceMock.swift
//  SatelliteIOS_Tests
//
//  Created by mingo on 2018/8/5.
//  Copyright © 2018年 CocoaPods. All rights reserved.
//

import Foundation
import CoreLocation
import SatelliteIOS

class LocationServiceMock : LocationService {
    
    var isStarted: Bool = false
    
    var lastLocation: CLLocation?
    var listener: SatelliteListener?
    var options: Satellite.Options?
    
    func start(options: Satellite.Options) {
        isStarted = true
        self.options = options
    }
    
    func stop() {
        isStarted = false
    }
    
    func setListener(_ listener: @escaping SatelliteListener) {
        self.listener = listener
    }
    
    func updateOptions(_ options: Satellite.Options) {
        self.options = options
    }
    
    func getOnce(options: Satellite.Options, listener: @escaping SatelliteListener) {
        fireLocation(once: true)
        listener(nil, lastLocation)
    }
    
    func fireLocation(once: Bool = false) {
        if !isStarted && !once {
            return
        }
        lastLocation = makeLocation()
        listener?(nil, lastLocation)
    }
    
    private func makeLocation() -> CLLocation {
        let location = CLLocation(coordinate: CLLocationCoordinate2D(latitude: 0, longitude: 0), altitude: 0, horizontalAccuracy: 0, verticalAccuracy: 0, timestamp: Date())
        return location
    }
    
}
